### PR TITLE
Create starter for s3 integration client

### LINKED ISSUE #28: Add a starter with the functionality to use digiwf-s3-integration on client side
**Is your feature request related to a problem? Please describe.**
This meant starter provides the client side capability to handle files with digiwf-s3-integration.

**Describe the solution you'd like**
This client provides the functionality to save, delete, update and get files from s3-storage via the digiwf-s3-integration starter.

A method call for file handling of the client starter has the following functionality:

1. Request the presignedURL from rest endpoints of digiwf-s3-integration starter.
2. Use the presignedURL to interact directly with s3 storage for file handling.

**Describe alternatives you've considered**
Each project has to code this client on its own.

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by boal**
close #28

### CODE REVIEW

File: digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/transfer/S3FileTransferRepository.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+     */
+    public byte[] getFile(final String presignedUrl) throws DocumentStorageClientErrorException, DocumentStorageServerErrorException, DocumentStorageException {
+        try {
+            final var headers = new HttpHeaders();
+            final HttpEntity<Void> httpEntity = new HttpEntity<>(headers);
+            /**
+             * Using the RestTemplate without any authorization.
+             * The presigned URL contains any authorization against the S3 storage.
+             */
+            final ResponseEntity<byte[]> responseEntity = new RestTemplate().exchange(
```
Comment (dominikhorn93) [inline digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/transfer/S3FileTransferRepository.java:42]: You have a RestTemplate Configuration where you provide a RestTemplate a a Bean. Why are you creating a new instance here?

File: digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/presignedurl/PresignedUrlRepository.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import org.springframework.web.client.RestClientException;
+
+import java.time.LocalDate;
+
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+public class PresignedUrlRepository {
+
+    private final FileApiApi fileApi;
```
Comment (dominikhorn93) [inline digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/presignedurl/PresignedUrlRepository.java:23]: Do you know why the class has the name "FileApiApi"?
https://github.com/it-at-m/digiwf-s3-integration/blob/dev/digiwf-s3-integration/src/main/java/io/muenchendigital/digiwf/s3/integration/api/controller/FileController.java @Tag name ist just "FileApi"!?

File: digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/transfer/S3FileTransferRepository.java (hunk 1 above)
Comment (boal) [reply digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/transfer/S3FileTransferRepository.java:42]: For the presigned request I want to use a plain request without any default headers and without other preconfigured settings. If i use an existing bean, there is the possibility that some additional headers (e.g. Access Token, ...) are sent with the request.

File: digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/presignedurl/PresignedUrlRepository.java (hunk 2 above)
Comment (boal) [reply digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/presignedurl/PresignedUrlRepository.java:23]: The open api generator ads always the postfix "Api" to each API class. Unfortunately this cannot be disabled by config or a manually defined postfix can be used.

Comment (dominikhorn93) [reply digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/presignedurl/PresignedUrlRepository.java:23]: So we should change the name in the @Tag Annotation? Does not belong to this PR... but maybe we should create another issue?

Comment (boal) [reply digiwf-s3-integration-client/src/main/java/io/muenchendigital/digiwf/s3/integration/client/repository/presignedurl/PresignedUrlRepository.java:23]: I think we should open an new issue. This problem needs some trial and error generation with open api.

**Review by dominikhorn93 — APPROVED** [review]

**Pull request merged**